server/controller/urlController.js

```diff
@@ -66,7 +68,48 @@ const createUrl = async (req, res) => {
   }
 };
 
+const signup = async (req, res) => {
+  try {
+    const { username, email, password } = req.body;
```

Comment:
Add input validation check for mandatory fields.

Reply:
Input validation means like chekcing whether the field is empty or not , am i correct?

Reply:
> Input validation means like chekcing whether the field is empty or not , am i correct?

Yes